frontend/src/components/RecordsList.tsx

```diff
@@ -42,6 +42,9 @@ interface IRecordsListProps {
   samplesRefetchWhereVariables: (
     samplesParsedSearchVals: string[]
   ) => SampleWhere;
+  sampleKeyForUpdate?: keyof Sample;
```

Comment:
`sampleKeyForUpdate` is a new prop/function argument that takes on the value of either "hasMetadataSampleMetadata" or "hasTempoTempos" and lets the component know where in the Sample data object it should make the update to.

"hasTempoTempos" is used in the Cohorts page, and "hasMetadataSampleMetadata" is used in all other pages.